### PR TITLE
Refine datanode major compaction

### PR DESCRIPTION
**Pull request by xiaocai2333**

### CODE REVIEW

File: internal/datacoord/garbage_collector.go [hunk 1]
```diff
@@ -566,3 +566,61 @@ func (gc *garbageCollector) recycleUnusedIndexFiles() {
 			zap.Int("delete index files num", deletedFilesNum))
 	}
 }
+
+func (gc *garbageCollector) recycleUnusedAnalyzeFiles() {
```
Comment (wayblink) [inline internal/datacoord/garbage_collector.go:570]: seems haven't added to work method

File: internal/datacoord/garbage_collector.go [hunk 2]
```diff
@@ -566,3 +566,61 @@ func (gc *garbageCollector) recycleUnusedIndexFiles() {
...
+	log.Info("start recycleUnusedAnalyzeFiles")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	startTs := time.Now()
+	prefix := path.Join(gc.option.cli.RootPath(), common.AnalyzeStatsPath) + "/"
+	// list dir first
+	keys, _, err := gc.option.cli.ListWithPrefix(ctx, prefix, false)
```
Comment (wayblink) [inline internal/datacoord/garbage_collector.go:577]: This list may cost much. Considering partition key collection may create partition_num analyze task in one major compaction, we may have hundreds or thousands analyze file at one time even Milvus all work well. Just a comment, it is long term problem. Or we can change the path pattern to optimize a bit？

Comment (xiaocai2333) [reply internal/datacoord/garbage_collector.go:578]: It sounds reasonable, but it feels more appropriate to put collectionID/partitionID before taskID.

File: internal/datacoord/analyze_meta.go [hunk 3]
```diff
@@ -168,3 +167,16 @@ func (m *analyzeMeta) GetAllTasks() map[int64]*indexpb.AnalyzeTask {
 
 	return m.tasks
 }
+
+func (m *analyzeMeta) CleanAnalyzeTask(taskID UniqueID) (bool, *indexpb.AnalyzeTask) {
```
Comment (czs007) [inline internal/datacoord/analyze_meta.go:171]: Better renamed as CheckCleanAnalyzeTask?

**Pull request merged**